x-pack/plugins/infra/public/pages/logs/log_entry_rate/sections/anomalies/expanded_row.tsx

```diff
@@ -4,86 +4,130 @@
-
+import React from 'react';
+import { useMount } from 'react-use';
 import { TimeRange } from '../../../../../../common/http_api/shared/time_range';
-import { AnalyzeInMlButton } from '../../../../../components/logging/log_analysis_results';
-import { LogEntryRateResults } from '../../use_log_entry_rate_results';
-import {
-  getAnnotationsForPartition,
-  getLogEntryRateSeriesForPartition,
-  getTotalNumberOfLogEntriesForPartition,
-} from '../helpers/data_formatters';
-import { AnomaliesChart } from './chart';
+import { AnomalyRecord } from '../../use_log_entry_rate_results';
+import { useLogEntryRateModuleContext } from '../../use_log_entry_rate_module';
+import { useLogEntryRateExamples } from '../../use_log_entry_rate_examples';
+import { LogEntryExampleMessages } from '../../../../../components/logging/log_entry_examples/log_entry_examples';
+import { bucketSpan } from '../../../../../../common/log_analysis/job_parameters';
+import { LogEntryRateExampleMessage, LogEntryRateExampleMessageHeaders } from './log_entry_example';
+import { euiStyled } from '../../../../../../../observability/public';
+
+const EXAMPLE_COUNT = 5;
+
+const examplesTitle = i18n.translate('xpack.infra.logs.analysis.anomaliesTableExamplesTitle', {
+  defaultMessage: 'Example logs',
+});
 
 export const AnomaliesTableExpandedRow: React.FunctionComponent<{
-  partitionId: string;
-  results: LogEntryRateResults;
-  setTimeRange: (timeRange: TimeRange) => void;
+  anomaly: AnomalyRecord;
   timeRange: TimeRange;
   jobId: string;
-}> = ({ results, timeRange, setTimeRange, partitionId, jobId }) => {
-  const logEntryRateSeries = useMemo(
-    () =>
-      results?.histogramBuckets ? getLogEntryRateSeriesForPartition(results, partitionId) : [],
-    /* eslint-disable-next-line react-hooks/exhaustive-deps */
-    [results, partitionId]
-  );
-  const anomalyAnnotations = useMemo(
-    () =>
-      results?.histogramBuckets
-        ? getAnnotationsForPartition(results, partitionId)
-        : {
-            warning: [],
-            minor: [],
-            major: [],
-            critical: [],
-          },
-    /* eslint-disable-next-line react-hooks/exhaustive-deps */
-    [results, partitionId]
-  );
-  const totalNumberOfLogEntries = useMemo(
-    () =>
-      results?.histogramBuckets
-        ? getTotalNumberOfLogEntriesForPartition(results, partitionId)
-        : undefined,
-    /* eslint-disable-next-line react-hooks/exhaustive-deps */
-    [results, partitionId]
-  );
+}> = ({ anomaly, timeRange, jobId }) => {
+  const {
+    sourceConfiguration: { sourceId },
+  } = useLogEntryRateModuleContext();
+
+  const {
+    getLogEntryRateExamples,
+    hasFailedLoadingLogEntryRateExamples,
+    isLoadingLogEntryRateExamples,
+    logEntryRateExamples,
+  } = useLogEntryRateExamples({
+    dataset: anomaly.partitionId,
+    endTime: anomaly.startTime + bucketSpan / 2,
+    exampleCount: EXAMPLE_COUNT,
+    sourceId,
+    startTime: anomaly.startTime - bucketSpan / 2,
```

Comment:
Hmm, yeah, possibly. I thought it might be more useful if the examples were "around" the anomaly time. But I guess that is inaccurate 🤔 I'll change this over to `(anomaly.startTime, anomaly.startTime + anomaly.duration)`.